### PR TITLE
54-55 group/event post routes

### PR DESCRIPTION
**Pull request by CB6670**
Server post routes to:

- Create a post 

- Get a specific post

- Get all posts related a user

- Get all posts related to an event 

- Get all posts related to group

### CODE REVIEW

File: server/apiRoutes/post.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        next(err);
+    }
+});
+
+//get all posts associated to a group
+router.get('/groupPosts', async (req, res, next) => {
+    try {
+        const posts = await Post.findAll({
+            where: {
+                groupId: req.params.groupId
```
Comment (CB6670) [inline server/apiRoutes/post.js:38]: I just wanted to make sure someone looks at these routes. I was wondering if I could pull a recent part of the URL here like this without explicitly stating it here (i.e. the req.params.groupId).

Comment (justin-cook-developer) [reply server/apiRoutes/post.js:38]: Nope. You need /groupPosts/:groupId

File: server/apiRoutes/post.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                eventId: req.params.eventId
+            }
+        })
+        res.json(posts);
+    } catch (err) {
+        next(err);
+    }
+});
+
+router.use((error, req, res, next) => {
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: This looks copy pasted and I don't see any custom errors in the file. Please remove this.

File: server/apiRoutes/post.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            }
+        });
+        res.json(posts);
+    } catch (err) {
+        next(err);
+    }
+});
+
+//get all posts associated to an event
+router.get('/eventPosts', async (req, res, next) => {
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: Need /eventPosts/:eventId

File: server/apiRoutes/post.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+            userId: req.user.id,
+            groupId: req.params.groupId,
+        }));
+    } catch (err) {
+        next(err);
+    }
+});
+
+// grab a single post i.e. when someone clicks on a post link to read comments
+router.get('/:postId', async (req, res, next) => {
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: This route needs to go below all other get/ routes. /:postId will catch anything and everything after the /.

File: server/apiRoutes/post.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                id: req.params.postId
+            }
+        }));
+    } catch (err) {
+        next(err)
+    }
+})
+
+//get all posts associated with a user
+router.get('/userPosts', async (req, res, next) => {
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: There should now be some isLoggedIn middleware in /utils/backend on the dev branch. please use that here.

**Review by justin-cook-developer — CHANGES_REQUESTED** [review]

File: server/apiRoutes/post.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
+const router = require('express').Router();
+// const { client, cacheDuration } = require('../redis');
+const { Post } = require('../db/index');
+const {isLoggedIn} = require('../../utils/backend');
+
+// create an event
+router.put('/createPost', isLoggedIn, async (req, res, next) => {
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: Make this a post route.

File: server/apiRoutes/post.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+    try {
+        res.json(await Post.create({
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: Split Create onto a seperate line.

File: server/apiRoutes/post.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        }));
+    } catch (err) {
+        next(err);
+    }
+});
+
+//get all posts associated with a user
+router.get('/userPosts', isLoggedIn, async (req, res, next) => {
+    try {
+        res.json(await Post.findAll({
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: Put queries on seperate line.

File: server/apiRoutes/post.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+            }
+        });
+        res.json(posts);
+    } catch (err) {
+        next(err);
+    }
+});
+
+//get all posts associated to an event
+router.get('/eventPosts/:eventId', async (req, res, next) => {
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: This route can be removed.

File: server/apiRoutes/post.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        res.json(posts);
+    } catch (err) {
+        next(err);
+    }
+});
+
+  // grab a single post i.e. when someone clicks on a post link to read comments
+router.get('/:postId', async (req, res, next) => {
+    try {
+        res.json(await Post.findOne({
```
Comment (justin-cook-developer) [inline server/apiRoutes/post.js]: Query on seperate line

**Review by justin-cook-developer — APPROVED** [review]

**Pull request merged**